### PR TITLE
refactor: supports babel 7

### PR DESCRIPTION
**Pull request by christophehurpeau**
Hello !

I tried this plugin and it didn't work with babel 7, because import is removed and binding lost.
The solution is to use `Program.exit` to remove it after.

Maybe you want to wait for the official babel 7 release before looking at this ?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,17 +38,18 @@
...
-    "babel-plugin-transform-es2015-modules-commonjs": "^6.24.1",
+    "@babel/plugin-syntax-jsx": "^7.0.0-beta.44",
+    "@babel/plugin-transform-modules-commonjs": "^7.0.0-beta.44",
     "classnames": "^2.2.5",
     "eslint": "^4.1.1",
     "eslint-config-prettier": "^2.3.0",
     "glob": "^7.1.2",
     "husky": "^0.14.2",
-    "jest": "^20.0.4",
+    "jest": "^22.4.3",
```
Comment (avaly) [inline package.json]: Can you please revert this `jest` upgrade? I think the new version introduces some formatting changes to the snapshot file. I prefer this PR to just upgrade `babel` and any requirements for  it.

I'll upgrade `jest` with a follow-up commit.

File: test/styles.css.js [hunk 2]
```diff
@@ -1,6 +1,4 @@
-const styles = {
+module.exports = {
```
Comment (avaly) [inline test/styles.css.js]: Any specific reason for changing this to commonjs?

**Review by avaly — CHANGES_REQUESTED** [review]
Thanks a lot for the changes! I would like to wait for non-beta v7 babel before merging this.

But please revert the `jest` for now.

File: test/styles.css.js (hunk 2 above)
Comment (christophehurpeau) [reply test/styles.css.js]: The tests fails without the change:
```
● executes basic.js

    /Users/chris/Work/utils/babel-plugin-inline-classnames/test/styles.css.js:6
    export default styles;
    ^^^^^^

    SyntaxError: Unexpected token export

      at ScriptTransformer._transformAndBuildScript (node_modules/jest-runtime/build/script_transformer.js:316:17)
      at eval (eval at it (test/index.test.js:30:18), <anonymous>:5:38)
      at Object.it (test/index.test.js:30:18)
```

This ones also fails:
- executes custom-import-name.js
- executes object-expression.js
- executes string-literal-mixed.js

File: package.json (hunk 1 above)
Comment (christophehurpeau) [reply package.json]: jest had a direct dependency to babel-core, witch they fixed in jest 21: https://github.com/facebook/jest/blob/master/CHANGELOG.md#jest-2100

> Change babel-core to peerDependency for compatibility with Babel 7

Comment (christophehurpeau) [reply package.json]: I can modify to 21 if you prefer

File: package.json [hunk 3]
```diff
@@ -38,17 +38,18 @@
...
+    "@babel/plugin-syntax-jsx": "^7.0.0-rc.3",
+    "@babel/plugin-transform-modules-commonjs": "^7.0.0-rc.3",
+    "babel-core": "^7.0.0-bridge.0",
+    "babel-eslint": "^8.2.6",
     "classnames": "^2.2.5",
     "eslint": "^5.4.0",
     "eslint-config-prettier": "^3.0.1",
     "glob": "^7.1.2",
     "husky": "^0.14.3",
-    "jest": "^23.5.0",
```
Comment (christophehurpeau) [inline package.json]: jest 23 is not compatible with babel 7: 
https://github.com/facebook/jest/pull/4923

Comment (christophehurpeau) [reply package.json]: issue: https://github.com/facebook/jest/issues/6913

File: package.json [hunk 4]
```diff
@@ -55,7 +57,7 @@
     "validate-commit-msg": "^2.14.0"
   },
   "peerDependencies": {
-    "babel-core": "6.*"
+    "babel-core": "6.* || ^7.0.0-bridge.0"
```
Comment (christophehurpeau) [inline package.json]: If you want to make a breaking change, I can update to `"@babel/core": "7.*"` ?

Comment (avaly) [reply package.json]: Yeah, let's do that. I'll publish this as a major version bump anyways.

Comment (christophehurpeau) [reply package.json]: done :)

File: package.json [hunk 5]
```diff
@@ -38,11 +38,13 @@
...
-    "babel-cli": "^6.24.1",
-    "babel-core": "^6.25.0",
-    "babel-eslint": "^7.2.3",
-    "babel-plugin-syntax-jsx": "^6.18.0",
-    "babel-plugin-transform-es2015-modules-commonjs": "^6.24.1",
+    "@babel/cli": "^7.0.0",
+    "@babel/core": "^7.0.0",
+    "@babel/plugin-syntax-jsx": "^7.0.0",
+    "@babel/plugin-transform-modules-commonjs": "^7.0.0",
+    "babel-core": "^7.0.0-bridge.0",
```
Comment (avaly) [inline package.json:45]: Let's remove this `babel-core`. It does not seem to have any usage.

File: package.json [hunk 6]
```diff
@@ -38,11 +38,13 @@
...
+    "babel-eslint": "^9.0.0",
+    "babel-jest": "^23.4.2",
```
Comment (avaly) [inline package.json:47]: Also, let's remove this from here. It should be included with `jest`.

File: package.json (hunk 5 above)
Comment (christophehurpeau) [reply package.json:45]: babel-jest uses it

File: package.json (hunk 6 above)
Comment (christophehurpeau) [reply package.json:47]: it is not, this is why tests were failing (https://github.com/avaly/babel-plugin-inline-classnames/pull/3#discussion_r212820321)

Comment (christophehurpeau) [reply package.json:47]: https://jestjs.io/docs/en/getting-started#using-babel

Comment (avaly) [reply package.json:47]: The following changes work fine for me on your branch: https://gist.github.com/avaly/f5760142157d618dfafcc2d27cce56ff

Comment (avaly) [reply package.json:47]: Ohh, my mistake. I had the files cached by jest. Indeed, it does not run without those packages.

But let's at least change the `babel-core` dependency to `babel-core@^7.0.0-0` according to the jest docs.

**Comment by christophehurpeau** [comment]
The documentation has the wrong name, there is a pr :
https://github.com/facebook/jest/pull/6745 🙂

**Comment by christophehurpeau** [comment]
Thanks !

**Pull request merged**